templates/commands/render.go

```diff
@@ -250,7 +254,14 @@ func (r *Render) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("os.Getwd(): %w", err)
 	}
 
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("os.UserHomeDir: %w", err)
+	}
+	backupDir := filepath.Join(homeDir, ".abc", "backups", strconv.FormatInt(time.Now().Unix(), 10))
```

Comment:
We might want to use:

```go
backupsDir := filepath.Join(homeDir, ".abc", "backups")
os.MkdirTemp(backupsDir, "*")
```

or: 

```go
now := time.Now().UTC().Unix()
backupsDir := filepath.Join(homeDir, ".abc", "backups")
os.MkdirTemp(backupsDir, fmt.Sprintf("%d-*", now))
```

Reply:
So, the reason to use `os.Mktempdir` is that it will _guarantee_ there's no conflict: https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/os/tempfile.go;l=80-111

In addition to the randomness and locking, it actually retries if there's a conflict.

I agree we shouldn't make the backup dir until it's needed; can we move this closer inline to when we know it'll be required then?

Reply:
Done. It's a little ugly because it's awkward to fake in tests.

If you want to go back to the original implementation which was simple but had a miniscule probabilty of collision in the case where two 64-bit random numbers collide within the same second, that's also good with me.